pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1308,7 +1308,7 @@ func (ctrl *PersistentVolumeController) provisionClaimOperation(claimObj interfa
 	tags[CloudVolumeCreatedForVolumeNameTag] = pvName
 
 	options := vol.VolumeOptions{
-		PersistentVolumeReclaimPolicy: v1.PersistentVolumeReclaimDelete,
+		PersistentVolumeReclaimPolicy: storageClass.ReclaimPolicy,
```

Comment:
yes, apiserver must do validation and give it defaults so it wont be possible for a storageclass to exist with a nil ReclaimPolicy